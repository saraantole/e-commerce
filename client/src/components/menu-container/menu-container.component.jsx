import MenuItem from '../menu-item/menu-item.component';

import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { selectMenuSections } from '../../redux/menu-container/menu-container.selectors';

import { MenuContainerDiv } from './menu-container.styles';

const MenuContainer = ({ menuSections }) => {
    return (
        <MenuContainerDiv>
            {
                menuSections.map(({ id, ...otherItemProps }) => (
                    <MenuItem key={id} {...otherItemProps} />
                )
                )
            }
        </MenuContainerDiv>
    )
}

const mapStateToProps = createStructuredSelector({
    menuSections: selectMenuSections
})

export default connect(mapStateToProps)(MenuContainer);