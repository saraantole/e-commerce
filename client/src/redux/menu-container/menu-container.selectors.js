import { createSelector } from 'reselect';

const selectMenuContainer = state => state.menuContainer;

export const selectMenuSections = createSelector(
    [selectMenuContainer],
    menuContainer => menuContainer.sections
);

