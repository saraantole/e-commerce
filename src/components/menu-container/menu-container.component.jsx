import MenuItem from '../menu-item/menu-item.component';

import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { selectMenuSections } from '../../redux/menu-container/menu-container.selectors';

import './menu-container.styles.scss';

const MenuContainer = ({ menuSections }) => {
    return (
        <div className='menu-container'>
            {
                menuSections.map(({ id, ...otherItemProps }) => (
                    <MenuItem key={id} {...otherItemProps} />
                )
                )
            }
        </div>
    )
}

const mapStateToProps = createStructuredSelector({
    menuSections: selectMenuSections
})

export default connect(mapStateToProps)(MenuContainer);