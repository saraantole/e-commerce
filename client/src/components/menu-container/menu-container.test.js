import React from 'react';
import { shallow } from 'enzyme';
import { MenuContainer } from './menu-container.component';

it('should render MenuContainer component', () => {
  expect(shallow(<MenuContainer menuSections={[]} />)).toMatchSnapshot();
});