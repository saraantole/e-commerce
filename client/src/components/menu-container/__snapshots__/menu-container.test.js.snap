// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MenuContainer component 1`] = `ShallowWrapper {}`;
